### PR TITLE
PHPStan (Level 1) on tests/

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | PHPStan (Level 0+1) on tests/
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Relative to  #16471 
| How to test?      | Github Actions are :green_circle: 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26592)

### CODE REVIEW

File: tests/Integration/PrestaShopBundle/Command/GenerateMailTemplatesCommandTest.php [hunk 1]
```diff
@@ -48,7 +48,7 @@ public function setUp(): void
 
     public function testMissingArguments()
     {
-        $this->expectException(RuntimeException::class, 'Not enough arguments (missing: "theme, locale").');
+        $this->expectException(RuntimeException::class);
```
Comment (matks) [inline tests/Integration/PrestaShopBundle/Command/GenerateMailTemplatesCommandTest.php:51]:
```suggestion
        $this->expectException(RuntimeException::class);
        $this->expectExceptionMessage('Not enough arguments (missing: "theme, locale").');
```
I would think this was the code that was supposed to be run.

File: tests/Unit/Core/Data/AbstractTypedCollectionTest.php [hunk 2]
```diff
@@ -48,7 +48,7 @@ public function testConstructor()
 
     public function testInvalidConstructor()
     {
-        $this->expectException(TypeException::class, 'Invalid element type Tests\Unit\Core\Data\InvalidCollectionTestElement, expected Tests\Unit\Core\Data\CollectionTestElement');
+        $this->expectException(TypeException::class);
```
Comment (matks) [inline tests/Unit/Core/Data/AbstractTypedCollectionTest.php:51]: Same here, I think a call `expectExceptionMessage` should make the assertion verified

File: tests/Unit/Core/Domain/ValueObject/MoneyTest.php [hunk 3]
```diff
@@ -44,8 +44,8 @@ public function testCreateWithPositiveValue()
 
     public function testCreateWithNegativeValue()
     {
-        $this->expectException(DomainConstraintException::class, 'Money amount cannot be lower than zero, -100 given');
```
Comment (matks) [inline tests/Unit/Core/Domain/ValueObject/MoneyTest.php:47]: Same here ▶️  `expectExceptionMessage`

File: tests/Unit/Core/MailTemplate/FolderThemeCatalogTest.php [hunk 4]
```diff
@@ -166,7 +166,7 @@ public function testGetByName()
 
     public function testInvalidTheme()
     {
-        $this->expectException(InvalidArgumentException::class, 'Invalid requested theme "unknown", only available themes are: classic, modern');
```
Comment (matks) [inline tests/Unit/Core/MailTemplate/FolderThemeCatalogTest.php:169]: Same here ▶️  `expectExceptionMessage`

**Review by matks — DISMISSED** [review]
Sorry @Progi1984 you need to rebase

**Comment by Progi1984** [comment]
@matks :heavy_check_mark: Done

**Review by matks — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thank you @Progi1984

**Pull request merged**